test/e2e/cert_manager_deployment_test.go

```diff
@@ -236,48 +236,62 @@ func TestCertRenew(t *testing.T) {
 			},
 		},
 	}
-	certmanagerv1Client, err := certmanagerclientset.NewForConfig(config)
-	require.NoError(t, err)
-	crt, err = certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Create(ctx, crt, metav1.CreateOptions{})
-	defer certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Delete(ctx, crt.ObjectMeta.Name, metav1.DeleteOptions{})
-	require.NoError(t, err)
-	err = wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+	// certmanagerClient is defined in utils_test and reusing it here for waitForCertificateReadiness().
+	certmanagerClient, err = certmanagerclientset.NewForConfig(config)
```

Comment:
Is the certmanagerClient overlapping with ginkgo and go test frameworks because of using the same package?

Reply:
Yeah, `waitForCertificateReadiness` makes use of the global parameter `certmanagerClient` and expects to be initialized when invoking it.

Reply:
It'd be better to use a local variable scoped specifically than the overlapping value.

Reply:
Yeah, but that would mean refactoring the code. For now I will create a local function where a variable is passed.